### PR TITLE
Give cards unique ids

### PR DESCRIPTION
**Pull request by granttitus**
All cards now have unique ideas. I modified only our GameState fixture, not the server code yet.

### CODE REVIEW

File: src/game/GameScene.ts [hunk 1]
```diff
@@ -349,11 +353,12 @@ class GameScene extends Phaser.Scene {
...
         const purgePileX = originX + playerDataOffsetX + 20 + nameWidth + CARD_WIDTH / 2
         const purgePileZone = this.createCardDropZone({
+            player,
             x: purgePileX,
             y: originY + CARD_HEIGHT / 2,
-            getCardImage: () => player.purged.length === 0 ? "cardback" : player.purged[player.purged.length - 1].id,
+            getCardImage: () => player.purged.length === 0 ? "cardback" : player.purged[player.purged.length - 1].backingCard.cardTitle,
             getMinimumAlpha: () => player.purged.length === 0 ? 0.1 : 1,
-            allowCardTypes: ["card-in-hand", "creature", "artifact", "upgrade"],
+            allowCardTypes: ["hand", "creature", "artifact", "upgrade"],
```
Comment (CorayThan) [inline src/game/GameScene.ts]: These should be an enum for safer reference and easier refactorability. 

```
export enum Cardtype {
  HAND = "hand",
  CREATURE = "creature",
  ARTIFACT = "artifact",
  UPGRADE = "upgrade"
}
```

Then functions like `createCardDropZone` and `getCardType` can be improved to be safely typed as well.

File: src/game/GameScene.ts [hunk 2]
```diff
@@ -502,8 +508,8 @@ class GameScene extends Phaser.Scene {
                 scene: this,
                 x: originX + CARD_WIDTH / 2 + i * Math.min((handWidth / player.hand.length), CARD_WIDTH * 0.8),
                 y: originY + CARD_HEIGHT / 2,
-                id: `${player.player.id}-card-in-hand-${i}`,
-                front: orientation === "top" ? "cardback" : player.hand[i].id,
+                id: player.hand[i].id,
+                front: orientation === "top" ? "cardback" : player.hand[i].backingCard.cardTitle,
```
Comment (CorayThan) [inline src/game/GameScene.ts]: Orientation should also be an enum for the same reasons.

File: src/game/GameScene.ts (hunk 1 above)
Comment (granttitus) [reply src/game/GameScene.ts]: Added a `CardType` enum

File: src/game/GameScene.ts (hunk 2 above)
Comment (granttitus) [reply src/game/GameScene.ts]: Done as well

**Pull request merged**